bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeBridgeHandler.java

```diff
@@ -0,0 +1,819 @@
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.AuthenticationStore;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.BasicAuthentication;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.util.component.AbstractLifeCycle;
+import org.eclipse.jetty.util.thread.QueuedThreadPool;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.openhab.binding.freeathomesystem.internal.configuration.FreeAtHomeBridgeHandlerConfiguration;
+import org.openhab.binding.freeathomesystem.internal.datamodel.FreeAtHomeDeviceDescription;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.stream.JsonReader;
+
+/**
+ * The {@link FreeAtHomeBridgeHandler} is responsible for handling the free@home bridge and
+ * its main communication.
+ *
+ * @author Andras Uhrin - Initial contribution
+ * @param <FreeAtHomeSystemHttpQueueResponse>
+ *
+ */
+@NonNullByDefault
+public class FreeAtHomeBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FreeAtHomeBridgeHandler.class);
+
+    public @Nullable static FreeAtHomeBridgeHandler freeAtHomeSystemHandler = null;
+
+    public ChannelUpdateHandler channelUpdateHandler = new ChannelUpdateHandler();
+
+    // Clients for the network communication
+    private HttpClient httpClient;
+    private @Nullable WebSocketClient websocketClient = null;
+    private @Nullable EventSocket socket = null;
+    private @Nullable FreeAtHomeWebsocketMonitorThread socketMonitor = null;
+    private @Nullable QueuedThreadPool jettyThreadPool = null;
+
+    private String sysApUID = "00000000-0000-0000-0000-000000000000";
+    private String ipAddress = "192.168.1.1";
+    private String username = "";
+    private String password = "";
+
+    private String baseUrl = "";
+
+    private String authField = "";
+
+    private AtomicBoolean httpConnectionOK = new AtomicBoolean(false);
+
+    int numberOfComponents = 0;
+
+    private static final int BRIDGE_WEBSOCKET_RECONNECT_DELAY = 60;
+
+    private List<String> listOfComponentId = new ArrayList<String>();
```

Comment:
Why instance variable, it seems this could be local within `getDeviceDeviceList`?

Reply:
understand... good idea. I changed